presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/TableScanMatcher.java

```diff
@@ -64,35 +63,8 @@ public MatchResult detailMatches(PlanNode node, PlanNodeCost cost, Session sessi
         String actualTableName = tableMetadata.getTable().getTableName();
         return new MatchResult(
                 expectedTableName.equalsIgnoreCase(actualTableName) &&
-                        domainMatches(tableScanNode, session, metadata));
-    }
-
-    private boolean domainMatches(TableScanNode tableScanNode, Session session, Metadata metadata)
-    {
-        if (!expectedConstraint.isPresent()) {
```

Comment:
It's a question of the meaning of the optional.  The factored out function now verifies that the optional item exists, or not, as specified.  The fact that each matcher uses an empty optional to indicate "don't care" is separate.